src/vmm/src/logger/metrics.rs

```diff
@@ -1146,6 +1061,27 @@ impl Serialize for SerializeToUtcTimestampMs {
     }
 }
 
+#[derive(Default, Debug)]
+// By using the below structure in FirecrackerMetrics it is easy
+// to serialise Firecracker app_metrics as a signle json object which
+// otherwise would have required extra string manipulation to pack
+// net as part of the same json object as FirecrackerMetrics.
+pub struct NetMetricsSerializer {}
+impl NetMetricsSerializer {
+    pub const fn new() -> Self {
+        Self {}
+    }
+}
```

Comment:
do you mean we use Default instead?

Reply:
If you have a unit struct in rust (e.g. without fields) you can just initialize it by referring to its name, e.g. 

```rs
pub struct NetMetricsProxy;

fn main() {
    let proxy = NetMetricsProxy;
    proxy.serialize(...)
}
```

Reply:
Sorry but I didn't understand the comment.
If the suggestion is to remove the NetMetricsProxy::new() then, we need the new() because FirecrackerMetrics::new() needs a const initialization which Default doesn't seem to have.
If the suggestion is not to have the NetMetricsProxy at all in FirecrackerMetrics then, the reason why this was added is because it is easier this way to pack NetDeviceMetrics inside FirecrackerMetrics as a single json object.

Reply:
What I meant is that you need neither new nor default, because in FirecrackerMetrics you can just use a struct initializer directly. Since its a unit struct, it can just be

impl FirecrackerMetrics {
    /// Const default construction.
    pub const fn new() -> Self {
        Self {
            ...
            net: NetDeviceMetrics {}, // or NetDeviceMetric,
            ...
        }
    }
}

definitely more of a nit though :)

Reply:
oh, sorry totally forgot about that. fixed!